services/app-api/src/postgres/contractAndRates/submitRate.ts

```diff
@@ -52,6 +52,9 @@ async function submitRate(
                         include: includeLatestSubmittedRateRev,
                     },
                 },
+                orderBy: {
+                    createdAt: 'desc',
+                },
```

Comment:
Interesting find here. When using the `rateID` to unlock a rate, `findFirst` did not find the first record in descending order. After forcing descending, it would then work correctly.